src/lib/logical_query_plan/compare_lqps.cpp

```diff
@@ -0,0 +1,291 @@
+#include "compare_lqps.hpp"
+
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/aggregate_node.hpp"
+#include "logical_query_plan/create_view_node.hpp"
+#include "logical_query_plan/delete_node.hpp"
+#include "logical_query_plan/drop_view_node.hpp"
+#include "logical_query_plan/dummy_table_node.hpp"
+#include "logical_query_plan/insert_node.hpp"
+#include "logical_query_plan/join_node.hpp"
+#include "logical_query_plan/limit_node.hpp"
+#include "logical_query_plan/logical_plan_root_node.hpp"
+#include "logical_query_plan/mock_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/projection_node.hpp"
+#include "logical_query_plan/show_columns_node.hpp"
+#include "logical_query_plan/show_tables_node.hpp"
+#include "logical_query_plan/sort_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "logical_query_plan/union_node.hpp"
+#include "logical_query_plan/update_node.hpp"
+#include "logical_query_plan/validate_node.hpp"
+
+namespace opossum {
+
+SemanticLQPCompare::SemanticLQPCompare(const std::shared_ptr<const AbstractLQPNode>& lhs, const std::shared_ptr<const AbstractLQPNode>& rhs): _lhs(lhs), _rhs(rhs) {
+
+}
+
+const std::vector<std::pair<std::shared_ptr<const AbstractLQPNode>, std::shared_ptr<const AbstractLQPNode>>>& SemanticLQPCompare::differing_subtrees() const {
+  return _differing_subtrees;
+}
+
+bool SemanticLQPCompare::operator()() {
+  if (!_structural_traverse(_lhs, _rhs)) {
+    return false;
+  }
+
+  return _semantical_traverse(_lhs, _rhs);
+}
+
+bool SemanticLQPCompare::_structural_traverse(const std::shared_ptr<const AbstractLQPNode>& lhs, const std::shared_ptr<const AbstractLQPNode>& rhs) {
```

Comment:
"traverse" is a command - how can it return true or false?

Reply:
Should be renamed to something along the lines of `lqp_type_structure_equals` or something, I agree.